x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/inference/action/GetInferenceModelAction.java

```diff
@@ -34,19 +34,40 @@ public GetInferenceModelAction() {
 
     public static class Request extends AcknowledgedRequest<GetInferenceModelAction.Request> {
 
+        private static boolean DEFAULT_DO_NOT_PERSIST_DEFAULT_CONFIGS = false;
```

Comment:
I don't understand why we need this as part of the user-facing request.

At first glance, this is needed to fix some issues with the test setup, which shouldn't lead to changes in production code.

Reply:
The option is internal to the transport action it cannot be set by an REST API user. The inference usage action however can be called by a REST API user. Xpack telemetry and monitoring also call the usage action.  If the inference API is not being used it is preferable not to create the .inference index, having the flag here stops any callers of the [GET /_xpack/usage](https://www.elastic.co/guide/en/elasticsearch/reference/current/usage-api.html) from indirectly creating the index. It is also very beneficial that this change fixes the tests but even for production code it should not be creating the index if a user calls the usage action.

This entire problem relates to the design decision to persist the default configs on read. The reasons for this design are:
- It is not known until late in the cluster startup lifecycle whether the linux x86 optimised model can be used
- The code should make that decision once and initialise the config with the correct model variant
- Doing this on read rather than at some fixed point in the cluster lifecycle means failures are trivially retried on the next read.

Reply:
Thanks, that explanation helps! And I missed that this is internal-only.